### ISSUE TITLE
Still duplication of frameworks

### ISSUE THREAD

**Issue by nightswimmings** [joel-costigliola/assertj-core#1714]
Now that we are finally getting rid of Hamcrest and making JUnit no dependant on third party libraries, looks like Junit team implemented their own Assertions (org.junit.jupiter.api.Assertions) so this is an step-back to finally get uniform testing.

Did AssertJ ever talked with Junit 5 team about the possibility of merging all the work in one of both sites alone? I mean I think we would all benefit if Junit removed Assertions and current functionality of it was adopted in AssertJ, or wither make AssertJ once for all, the JUnit integrated assertion framework.

Is this remotely feasible?

**Comment by nightswimmings** [comment]
I asked the same to Junit team on Gitter: https://gitter.im/junit-team/junit5

**Comment by sormuras** [comment]
> [...] JUnit no dependant on third party libraries [...]

No dependencies, except for...

- [OpenTest4J](https://github.com/ota4j-team/opentest4j)
- [APIGuardian](https://github.com/apiguardian-team/apiguardian)
- JUnit Platform (okay, not strictly **3rd**-party, but... still a dependencies with multiple artifacts)


From the OpenTest4J README:
> There is no standard for testing on the JVM: the only common building block we have is `java.lang.AssertionError`.

Assertion frameworks are built around `java.lang.AssertionError`, providing better feedback on errors then just: _something failed_.

But what is "better"?

Read the lines provided by Google [Truth](https://truth.dev/):
> ## Why use an assertion library?
> In other words, why depend on a new library when you can use the methods built into your testing framework, like assertEquals?
> ...
> ### Truth vs. AssertJ
> ...
> ### Truth vs. Hamcrest
> ...

In short: "better" is always ... a different thing.

Thus, we as the JUnit team provide and use our `Assertions`. But we also like and use other assertion frameworks. We actually encourage users to try and use 3rd-party frameworks: https://junit.org/junit5/docs/current/user-guide/#writing-tests-assertions-third-party

**Comment by nightswimmings** [comment]
I meant third party libraries in terms of assertion framework. I understand Java built-in is not enough standard, but Junit is the core library imported for all tests, so a JUnit interface for assertions could serve for Hamcrest, AssertJ etc... that way we can exclude the built-in assertion implementation of the very same JUnit, in worst scenario, so we can force uniform declarations accross project

**Comment by joel-costigliola** [comment]
I think the main reason for having separate assertions library is to allow different release frequency.
Moreover you have people still using TestNG that can benefit from third party assertions library.

**Comment by nightswimmings** [comment]
To be honest, AssertionJ is impossible to match on completeness, eye-candiness, etc.. (kudos!), I am simply so desperate to make it the reference standard some way

**Comment by joel-costigliola** [comment]
Thanks for the kind words @nightswimmings! Anything to add or shall I close this issue?

**Comment by nightswimmings** [comment]
I guess it was more a discussion than a question so feel free to close it :'(
